configuration.md

```diff
@@ -0,0 +1,127 @@
+# Proposal for configuring supported
+
+Teams may want to customize the parameters of a support policy. For instance, a team may want to ignore certain dependencies, i.e. not enforce upgrades for certain dependencies. They may also want different upgrade cadences for specific dependencies. Lastly, in many cases, simply turning on a support policy and expecting consumers to immediately adhere to the policy is untenable (for instance, a major upgrade with a migration may be eminently required shortly after turning on a support policy). There needs to be a way to provide a grace period for onboarding to the support policy, while still providing a warning for when dependencies must be updated.
+
+We will not require a configuration file, and will default to 4, 2, and 1 quarter upgrade periods for major, minor, and patch versions respectively.
+
+```JSON
+{
+  "primary": {
+    "upgradeBudget": {
+      "major": 4,
+      "minor": 4,
+      "patch": 2
+    }
+  }
+}
+```
+
+```TypeScript
+type Quarter = number;
+type Weeks = number;
+type DateString = string // month/day/year for instance "12/25/2021"
+
+interface UpgradeBudget {
+  major: Quarters;
+  minor: Quarters;
+  patch: Quarters;
+
+  // Any releases that happen within this number of weeks of the end of a quarter
+  // are considered to have been released the next quarter
+  tailGracePeriod?: Weeks;
```

Comment:
Do we have a specific reason why we are going with weeks? as all other calculations are qtrs or months